### PR TITLE
feat: add counters to propagation summary

### PR DESCRIPTION
**Pull request by asalzburger**
This PR adds statistics counters for sensitives, portals and material surfaces to the Propagation summary and adds a little script to plot those.

ITK-sensitive-modules

It also adapts the `geomodel.py` to recent changes to be able to make this plot.

Changes to references are expected.

### CODE REVIEW

**Comment by asalzburger** [comment]
> This could be generalized with #3450 but I was not able to follow up with that yet

Yes, I actually like the split between "Summary" and "Step" writing, so we have small files for overview checks and big ones for detailed checks.

File: Examples/Scripts/Propagation/propagation_summary.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    "-o",
+    "--output",
+    type=str,
+    default="",
+    help="Output file base name",
+)
+
+args = p.parse_args()
+
+try:
```
Comment (benjaminhuth) [inline Examples/Scripts/Propagation/propagation_summary.py:74]: Hmm this swallows any kind of error, also errors not related to files and labels...
cant we just `assert len(args.input) == len(args.label) == len(args.marker)` to ensure this?

File: Examples/Scripts/Propagation/propagation_summary.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    fig, ax = plt.subplots(1, 1, figsize=(11, 10))
+
+    eta_bins = np.linspace(-4, 4, 100)
+    eta_centers = 0.5 * (eta_bins[:-1] + eta_bins[1:])
+    eta_width = 8.0 / 100
+
+    for ir, irfile in enumerate(args.input):
```
Comment (benjaminhuth) [inline Examples/Scripts/Propagation/propagation_summary.py:82]: We could be a bit nicer with 
```suggestion
    for irfile, label, marker in zip(args.input, args.label, args.marker):
```

**Review by benjaminhuth — COMMENTED** [review]
Looks good, some small nitpicky comments

**Review by benjaminhuth — DISMISSED** [review]

**Comment by asalzburger** [comment]
As discussed, I deleted the file.

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [e59014066cc1830cd5d42d343f1f5f6e86c452c7]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43413112) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/8103088)!

**Pull request merged**